# Publications

[Kirkegaard et al 2016: "Genomic insights into members of the candidate phylum Hyd24-12 common in mesophilic anaerobic digesters" DOI: 10.1038/ismej.2016.43](https://doi.org/10.1038/ismej.2016.43)
* Data availability
  * The raw metagenome sequencing data is available at the European Nucleotide Archive with BioProject ID: [PRJEB10929](http://www.ncbi.nlm.nih.gov/bioproject/?term=PRJEB10929).
  * The data processing is described in the [paper](https://doi.org/10.1038/ismej.2016.43) and the necessary workflow scripts are available at Github from the [mmgenome package](https://github.com/MadsAlbertsen/mmgenome). The core idea behind the genome extraction using differential coverage binning was published by [Albertsen et al 2013](https://doi.org/10.1038/nbt.2579)
  * The ".RData" files used for binning of genomes from metagenome sequencing data with the [mmgenome package](http://madsalbertsen.github.io/mmgenome/) is available at [figshare](https://figshare.com/authors/Rasmus_Kirkegaard/2604133): [Metagenomes from Randers](https://doi.org/10.6084/m9.figshare.3168220), [Metagenomes from Viborg](https://doi.org/10.6084/m9.figshare.3168229), [Metagenomes from Damhusaaen](https://doi.org/10.6084/m9.figshare.3168382).
  * RMarkdown files for reproducing the genome binning process are available at Github: [Kirkegaard et 2016](https://github.com/Kirk3gaard/Publications/tree/master/Kirkegaard2016)
  * The genome sequence data have been submitted to DDBJ/EMBL/GenBank databases under accession numbers [LKHB00000000](http://www.ncbi.nlm.nih.gov/nuccore/LKHB00000000), [LKHC00000000](http://www.ncbi.nlm.nih.gov/nuccore/LKHC00000000) and [LKHD00000000](http://www.ncbi.nlm.nih.gov/nuccore/LKHD00000000).

[Kirkegaard et al 2017: "Identifying the abundant and active microorganisms common to full scale anaerobic digesters"]()
* Data availability
  * The raw sequencing data is available at the European Nucleotide Archive with BioProject ID: [PRJEB15624](http://www.ncbi.nlm.nih.gov/bioproject/?term=PRJEB15624).
  * The data processing is described in the [paper]()
  * The ".RData" file is available at [figshare](https://dx.doi.org/10.6084/m9.figshare.4308191)
  * RMarkdown files for reproducing the figures are available at Github: [Kirkegaard et 2017](https://github.com/Kirk3gaard/Publications/tree/master/Kirkegaard2017)
