---
title: "Identifying the abundant and active microorganisms common to full-scale anaerobic digesters"
author: "Rasmus Kirkegaard et al"
date: "`r format(Sys.time(), '%d-%m-%Y')`, Aalborg, Denmark"
output: html_document
---

```{r , message=FALSE, warning=FALSE, echo=F}
library(ampvis)
library(cowplot)
library(plyr)
library(ggrepel)
library(plotly)
library(gridExtra)
library(grid)
```

## Load data

```{r echo=T}
status<-try(load("data.RData"),silent = T)
if(class(status)=="try-error") {
  rm(list = ls())
  print("Loading data from scratch")
  knit("Load_data.Rmd")
}
```

Print sessioninfo for reproducibility as R and package versions are known to make a difference (and occasionally conflicts).

```{r}
sessionInfo()
```

# Analyse the microbial data

## Bacteria
The bacterial part of the community has been sequenced using primers covering the V1-V3 region of the 16S rRNA gene. (27F (AGAGTTTGATCCTGGCTCAG, Lane 1991) and 534R (ATTACCGCGGCTGCTGG, Muyzer et al, 1993))

### Subset and filtering
The principle behind ampvis is that you first subset the data to what you want to look at using `phyloseq` and then visualise it using `ampvis`. Samples can be subset based on any available metadata. See the [phyloseq guide](http://joey711.github.io/phyloseq/tutorials-index) for more examples.

```{r see_variables, eval=F}
sample_variables(b_d)
```

Before we analyse the data it is often convenient to remove low abundant OTUs, as they are too noisy to result in meaningful conclusions. This can be done using the `filter_taxa` phyloseq function. Here we keep OTUs that have been seen more than 9 times (of 10000) in at least 1 sample (Keeping OTUs with an abundance of 0.1 % or more).

```{r}
b_transformed<-transform_sample_counts(b_d, function(x) x / sum(x) * 100)
b_df <- filter_taxa(b_transformed, function(x) max(x) >= 0.1, TRUE)
remove(b_transformed,b_d)
```

```{r}
foamsamples<-c("SA-RHK-527",
               "SA-RHK-529",
               "SA-RHK-531",
               "SA-RHK-1052",
               "SA-RHK-1053",
               "SA-RHK-1054",
               "SA-RHK-1055",
               "SA-RHK-1056",
               "SA-RHK-1075",
               "SA-RHK-1243",
               "SA-RHK-1244",
               "SA-RHK-1302",
               "SA-RHK-1339"
               )
```


# Add a new set of variables
To avoid introducing more copy/paste errors in microsoft excel additional metadata variables are created here

```{r}
# Make a variable
sample_data(b_df)$simple_var<-"empty"
sample_data(b_df)$simple_var[which(get_variable(b_df, "Sample_type") %in% c("WWTP-Sludge","Procestank"))]<-"Surplus_sludge"
sample_data(b_df)$simple_var[which(get_variable(b_df, "Sample_type") %in% c("Before_primary_settling","Influent"))]<-"Before_primary_settling"
sample_data(b_df)$simple_var[which(get_variable(b_df, "Sample_type") %in% c("After_primary_settling"))]<-"After_primary_settling"
sample_data(b_df)$simple_var[which(get_variable(b_df, "Sample_type") %in% c("Effluent"))]<-"Effluent"
sample_data(b_df)$simple_var[which(get_variable(b_df, "Sample_type") %in% c("Digester"))]<-"Digester"
sample_data(b_df)$simple_var[which(get_variable(b_df, "Temperature") %in% "Thermophilic")]<-"Thermophilic"
sample_data(b_df)$simple_var[which(get_variable(b_df, "Temperature") %in% "Mesophilic")]<-"Mesophilic"
sample_data(b_df)$simple_var[which(get_variable(b_df, "Temperature") %in% "THP")]<-"THP"

# Make a variable
sample_data(b_df)$simple_var2<-"empty"
sample_data(b_df)$simple_var2[which(get_variable(b_df, "Sample_type") %in% c("WWTP-Sludge","Processtank"))]<-"Surplus_sludge"
sample_data(b_df)$simple_var2[which(get_variable(b_df, "Sample_type") %in% c("Effluent"))]<-"Effluent"
sample_data(b_df)$simple_var2[which(get_variable(b_df, "Sample_type") %in% c("Before_primary_settling","Influent"))]<-"Primary_sludge" # 20160913: MNI & JMK told me that "After_primary_settling" is not a part of the primary sludge
sample_data(b_df)$simple_var2[which(get_variable(b_df, "Temperature") %in% "Thermophilic")]<-"Thermophilic"
sample_data(b_df)$simple_var2[which(get_variable(b_df, "Temperature") %in% "Mesophilic")]<-"Mesophilic"
sample_data(b_df)$simple_var2[which(get_variable(b_df, "Temperature") %in% "THP")]<-"THP" 

# Make a variable
sample_data(b_df)$influent_or_ad<-"empty"
sample_data(b_df)$influent_or_ad[which(get_variable(b_df, "simple_var") %in% c("Before_primary_settling","Surplus_sludge","Influent"))]<-"a) Influent" # # 20160913: MNI & JMK told me that "After_primary_settling" is not a part of the primary sludge
sample_data(b_df)$influent_or_ad[which(get_variable(b_df, "simple_var") %in% c("Thermophilic","Mesophilic","THP","Digester"))]<-"b) Digester"
sample_data(b_df)$influent_or_ad[which(get_variable(b_df, "simple_var") %in% c("Effluent"))]<-"c) Effluent"

# Make a variable
sample_data(b_df)$simple_var3<-"empty"
sample_data(b_df)$simple_var3[which(get_variable(b_df, "simple_var") %in% c("Before_primary_settling","Influent"))]<-"a) Primary sludge" # 20160913: MNI & JMK told me that "After_primary_settling" is not a part of the primary sludge
sample_data(b_df)$simple_var3[which(get_variable(b_df, "simple_var") %in% c("Surplus_sludge"))]<-"b) Surplus sludge"
sample_data(b_df)$simple_var3[which(get_variable(b_df, "simple_var") %in% c("Mesophilic"))]<-"c) Mesophilic"
sample_data(b_df)$simple_var3[which(get_variable(b_df, "simple_var") %in% c("Thermophilic"))]<-"d) Thermophilic"
sample_data(b_df)$simple_var3[which(get_variable(b_df, "simple_var") %in% c("THP"))]<-"e) THP"

# Make a variable
sample_data(b_df)$order_type<-"empty"
sample_data(b_df)$order_type[which(get_variable(b_df, "simple_var") %in% c("Before_primary_settling","Influent"))]<-"1_primary" # 20160913: MNI & JMK told me that "After_primary_settling" is not a part of the primary sludge
sample_data(b_df)$order_type[which(get_variable(b_df, "simple_var") %in% c("Surplus_sludge"))]<-"2_surplus"
sample_data(b_df)$order_type[which(get_variable(b_df, "simple_var") %in% c("Mesophilic","Thermophilic","THP","Digester"))]<-"3_digester"
sample_data(b_df)$order_type[which(get_variable(b_df, "simple_var") %in% c("Effluent"))]<-"4_effluent"

# Make a variable
sample_data(b_df)$Foamevent<-"normal"
sample_data(b_df)$Foamevent[which(get_variable(b_df, "SA_ID") %in% foamsamples)]<-"foam"
```

Here we will use subset to select samples by variables in the metadata and store the data in the object b_ads.

```{r}
b_ads<- subset_samples(b_df, Sample_location %in% c("AAV","Viborg","Fredericia","Damhusaaen","Randers","Herning","Fornaes","AAoe","aaby","Hjoerring","Soeholt","Slagelse","Ejby_Moelle","Bjergmarken","Lundtofte","Hobro","Avedoere","Naestved","Ringkoebing", "Esbjerg Vest") & Temperature %in% c("THP","Mesophilic","Thermophilic") & !(Project_name %in% c("DNAextraction_optimisation_from_biogas","VTU_Case_study_-_GasMixCase_Viborg")))
```

Lets have a look at the data

```{r, fig.height=8,fig.width=8,fig.align="center"}
amp_ordinate(data = b_ads)+theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))
```

Let us add some information to see if we can see any patterns

```{r, fig.height=10,fig.width=12,fig.align="center"}
amp_ordinate(data = b_ads,
             plot.label = "SEQ_ID",
             plot.color = "Temperature",
             #plot.group.label = "Temperature",
             #plot.group.label.size = 7,
             plot.color.order = c("THP","Thermophilic","Mesophilic"))+theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))
```

There seems to be somewhat defined microbial communities with regard to the process types: Mesophilic, Thermophilic and THP. However, some of the samples seem to be clustering with the wrong groups. So I did some microbial detective work. 

* It turns out that Aaby was actually operating at mesophilic temperatures in 2011 even though they reported otherwise. 
* The mesophilic reactor in Herning is connected in series with the thermophilic reactor as influent, and is thus not representative for a "standard" mesophilic reactor.

It thus seems fair to remove these and samples like it for now.

```{r}


skiplist<-c(foamsamples,"SA-RHK-921", "SA-RHK-1028","SA-RHK-1065","SA-RHK-1311","SA-RHK-19","SA-RHK-212")
#"SA-RHK-1028" (Sample from aaby 2011, where the plant was actually run with a mesophilic process)
#"SA-RHK-1065" (Sample from aaby 2011, where the plant was actually run with a mesophilic process)
#"SA-RHK-1311" (Sample from aaby 2015, what are they doing now???)
#"SA-RHK-921" (Sample from Herning when there was no info about reactor no)
#"Herning_M" is connected in series to a thermophilic plant and is thus not representative for a mesophilic plant 
skiplist2<-c(paste0("16SAMP-",9094:9147),paste0("16SAMP-",7593:7614),paste0("16SAMP-",8858:8867),paste0("16SAMP-",8984:8995),paste0("16SAMP-",8902:8903),"16SAMP-3461","16SAMP-3455","16SAMP-11802","16SAMP-11829","16SAMP-3464")

b_ads_s<- subset_samples(b_ads, !(SA_ID %in% skiplist) & !(Sample_location=="Herning" & Temperature=="Mesophilic") & (Investigator!="CHJ") & !(SEQ_ID %in% skiplist2) & !(Dataset=="CJH" & Temperature=="THP"))
```

With the filtered dataset we can now proceed and prepare the figures in the paper

# Main figures

## Figure 1
Panelfigure with 4 PCA plots. 

a. Bacteria separates into 3 clusters
b. Archaea separates by temperature
c. Mesophilic bacteria cluster by plant
d. Thermophilic bacteria cluster by plant

### Figure 1a

```{r}
p1_a<-amp_ordinate(data = b_ads_s,
             plot.color = "Temperature",
             plot.group.label = "Temperature",
             plot.group.label.size = 7,
             plot.color.order = c("THP","Thermophilic","Mesophilic")
             )+theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))+theme(legend.position="none")
```

Now the groups of THP, Mesophilic, and Thermophilic are a lot more well defined. We can also have a look at the underlying plant separation.

### Figure 1c

How do the samples group within the Mesophilic cluster?

```{r, fig.height=10,fig.width=12, echo=F,fig.align="center"}
b_ads_sM<- subset_samples(b_ads_s, Temperature=="Mesophilic" & Project_name!="VTU Case study - GasMixCase Viborg") # The Gasmix case is removed to avoid the main difference to be between Viborg and other plants due to uneven sample number.
p1_c<-amp_ordinate(data = b_ads_sM,
             plot.color = "Sample_location",
             plot.group = "chull",
             plot.group.label = "Sample_location"
             )+theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))+theme(legend.position="none")
```

### Figure 1d

How do the samples group within the Thermophilic cluster?

```{r, fig.height=10,fig.width=12, echo=F,fig.align="center"}
b_ads_sT<- subset_samples(b_ads_s, Temperature=="Thermophilic")
p1_d<-amp_ordinate(data = b_ads_sT,
             plot.color = "Sample_location",
             plot.group = "chull",
             plot.group.label = "Sample_location"
             )+theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))+theme(legend.position="none")
```

## Archaea
The Archaeal part of the community has been sequenced using primers covering the V3-V5 region of the 16S rRNA gene. (340F (CCCTAHGGGGYGCASCA, Pinto et al., 2012) and 915R (GWGCYCCCCCGYCAATTC, Pinto et al., 2012))

```{r}
a_df <- filter_taxa(a_d, function(x) max(x) >= 0.1, TRUE)
amp_export(a_df,file = "archaea_filter.fa",tax = T)
```


```{r}
a_ads<- subset_samples(a_df, Sample_location %in% c("AAV","Viborg","Fredericia","Damhusaaen","Randers","Herning","Fornaes","AAoe","aaby","Hjoerring","Soeholt","Slagelse","Ejby Moelle","Bjergmarken","Lundtofte","Hobro","Avedoere","Naestved","Ringkoebing","Esbjerg Vest") & Temperature %in% c("THP","Mesophilic","Thermophilic") & !(Project_name %in% c("DNAextraction optimisation from biogas","VTU Case study - GasMixCase Viborg")))
```

```{r , fig.height=10,fig.width=12, echo=F,fig.align="center"}
# amp_ordinate(data = a_ads,
#              plot.color = "Temperature"
#              )+theme_cowplot()
```

The Archaea shows a clear cut between thermophilic reactors and mesophilic reactors (The THP reactors do also have a mesophilic process). Removing the samples that were identified to be wrongly classified before will make this trend even clearer.

```{r}
a_ads_s<- subset_samples(a_ads, !(SA_ID %in% skiplist) & !(Sample_location=="Herning" & Temperature=="Mesophilic") & (Investigator!="CHJ"))
```

### Figure 1b

```{r,fig.align="center"}
p1_b<-amp_ordinate(data = a_ads_s,
             plot.color = "Temperature",
             plot.group.label = "Temperature",
             plot.group.label.size = 7,
             plot.color.order = c("THP","Thermophilic","Mesophilic"))+theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))+theme(legend.position="none")
```

### Figure 1 combined

```{r,fig.align="center",fig.height=10,fig.width=10}
plot_grid(p1_a, p1_b, p1_c, p1_d, align='vh',ncol = 2, labels=c('A', 'B', 'C', 'D'))
#ggsave(filename = "SurveyPaper/Figure1.pdf",width = 10,height = 10, useDingbats = FALSE)
```


## Figure 2
Panelfigure with 2 heatmaps. 

a. Phylum level heatmap for bacteria
b. Genus level heatmap for bacteria

### Figure 2a

Phylum level heatmap for bacteria

```{r,fig.align="center",fig.width=15}
p2_a<-amp_heatmap(b_ads_s,tax.aggregate = "Phylum",tax.show = 20,group = c("Temperature","Sample_location"),color.vector = c("white","red"),plot.na = T)+
  facet_grid(~ Temperature,scales="free_x",space="free")+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")

# Change labels
original_labels<-levels(p2_a$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p2_a$data$Group.label<-mapvalues(p2_a$data$Group,original_labels,c(new_labels)) 
p2_a<-p2_a+aes(x=p2_a$data$Group.label)+theme(axis.title.x=element_blank())
```

### Figure 2b
Heatmap of top 25 genera in anaerobic digesters

```{r,fig.align="center",fig.height=9,fig.width=15}
p2_b<-amp_heatmap(b_ads_s,tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = 20,group = c("Temperature","Sample_location"),color.vector = c("white","red"),plot.na = T)+
  facet_grid(~ Temperature,scales="free_x",space="free")+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")
# Change labels
original_labels<-levels(p2_b$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p2_b$data$Group.label<-mapvalues(p2_b$data$Group,original_labels,c(new_labels)) 
p2_b<-p2_b+aes(x=p2_b$data$Group.label)+theme(axis.title.x=element_blank())
```


### Figure 2 combined

```{r ,fig.align="center",fig.width=18, fig.height=13}
plot_grid(p2_a, p2_b, align='vh',ncol = 1, labels=c('A', 'B'),rel_heights = c(1,1))
#ggsave(filename = "SurveyPaper/Figure2.pdf",width = 18,height = 13, useDingbats = FALSE)
```

Get stats about the number of samples pr plant

```{r}
mytable<-sample_data(b_ads_s)
summary(mytable$Sample_location)
```

## Figure 3
Panelfigure with 2 heatmaps. 

a. Phylum level heatmap for Archaea
b. Genus level heatmap for Archaea

### Figure 3a

Phylum level heatmap for archaea

```{r ,fig.align="center",fig.height=3,fig.width=15}
p3_a<-amp_heatmap(a_ads_s,tax.aggregate = "Phylum",tax.show = 5,group = c("Temperature","Sample_location"),color.vector = c("white","red"),plot.na = T)+
  facet_grid(~ Temperature,scales="free_x",space="free")+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")

# Change labels
original_labels<-levels(p3_a$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p3_a$data$Group.label<-mapvalues(p3_a$data$Group,original_labels,c(new_labels)) 
p3_a<-p3_a+aes(x=p3_a$data$Group.label)+theme(axis.title.x=element_blank())
```

### Figure 3b

Genus level heatmap for Archaea
```{r,fig.align="center",fig.width=15}
p_temp<-amp_heatmap(a_ads_s,tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = 20,group = c("Temperature","Sample_location"),color.vector = c("white","red"),plot.na = T)+
  facet_grid(~ Temperature,scales="free_x",space="free")+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")

# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
p3_b<-p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank())
```

### Figure 3 combined

```{r ,fig.align="center",fig.width=18, fig.height=13}
plot_grid(p3_a, p3_b, align='vh',ncol = 1, labels=c('A', 'B'),rel_heights = c(4,9))
#ggsave(filename = "SurveyPaper/Figure3.pdf",width = 18,height = 13, useDingbats = FALSE)
```


Get stats about the number of samples pr plant

```{r}
mytable<-sample_data(a_ads_s)
summary(mytable$Sample_location)
```

## Figure 4
Digester abundance for the OTUs versus the ratio between the abundance in the digester compared to the abundance in the finluent streams (primary and surplus sludge)

 Abundance data for ratios OTU level
```{r}
reactors<-c("Mesophilic","Thermophilic","THP")
ratiodf<-data.frame(RATIO=double(), res_RT.tag=character(), res_RT.Mean=double(), T_reactor=character())
ratiodf_h<-data.frame(init=1337)
for (T_reactor in reactors) {

####
# Digester
###
d_sel<- subset_samples(b_ads_s, Temperature==T_reactor)

res<-data.frame("tag"=names(apply(otu_table(d_sel), 1, mean)),"Mean"=apply(otu_table(d_sel), 1, mean))

res_RT<-res

####
# For influent
###
d_sel<- subset_samples(b_df, Sample_type=="Influent")

res<-data.frame("tag"=names(apply(otu_table(d_sel), 1, mean)),"Mean"=apply(otu_table(d_sel), 1, mean))

res_inf<-res
####
# Surplus sludge
###
d_sel<- subset_samples(b_df, simple_var3=="b) Surplus sludge")

res<-data.frame("tag"=names(apply(otu_table(d_sel), 1, mean)),"Mean"=apply(otu_table(d_sel), 1, mean))

res_AS<-res

# Calculate ratio and plot
options(scipen = 8)
RATIO_inf<-res_RT$Mean/res_inf$Mean
RATIO_AS<-res_RT$Mean/res_AS$Mean
RATIO<-apply(cbind(RATIO_inf,RATIO_AS),1,min)

mydf<-data.frame(RATIO,res_RT$tag,res_RT$Mean)


# For creating a table of all the abundance values
tempdf<-cbind(mydf,T_reactor)
tempdf$Frequency<-tempdf$res_RT.Mean/sum(tempdf$res_RT.Mean)
ratiodf<-rbind(ratiodf,tempdf)
ratiodf_h<-cbind(ratiodf_h,mydf)
}
```


```{r fig.height=10,fig.width=10}
ratiodf_lim<-ratiodf[which(ratiodf$res_RT.Mean>2),]
ratiodf_lim<-ratiodf_lim[order(ratiodf_lim$res_RT.Mean,decreasing = T),]
p<-ggplot(ratiodf_lim, aes(x = RATIO, y = res_RT.Mean, colour=T_reactor))+
  geom_point()+scale_x_log10(breaks=c(0.01,10,10000),labels = c(0.01,10,10000))+geom_vline(xintercept=1,linetype = "longdash", colour = "grey")+ylab("Mean abundance in AD")+geom_text_repel(aes(x = ratiodf_lim$RATIO,
                      y = ratiodf_lim$res_RT.Mean,
                      label = ratiodf_lim$res_RT.tag)) 
p+geom_point(data=ratiodf,aes(x = RATIO, y = res_RT.Mean))+guides(colour = guide_legend(title = "Reactor type",override.aes = list(size = 3)))+theme(legend.key=element_rect(fill=NA),legend.position="bottom")
fname<-paste0("SurveyPaper/Figure4.pdf")
#ggsave(filename = fname,width = 10,height = 10, useDingbats = FALSE)



seltax<-taxinfo[which(rownames(taxinfo)%in%unique(ratiodf_lim$res_RT.tag)),]
tempnames<-rownames(seltax)
seltax <- lapply(seltax, function(x) {gsub("p__", "", x)})
seltax <- lapply(seltax, function(x) {gsub("c__", "", x)})
seltax <- lapply(seltax, function(x) {gsub("o__", "", x)})
seltax <- lapply(seltax, function(x) {gsub("f__", "", x)})
seltax<-as.data.frame(seltax,row.names = tempnames)
grid.arrange(
tableGrob(seltax[as.character(unique(ratiodf_lim$res_RT.tag)),c("Phylum","Class","Order","Family","Genus")])
)
pdf(file = "SurveyPaper/Figure4_OTUlabels.pdf",width = 10, height = 5.5)#10*10
grid.table(seltax[as.character(unique(ratiodf_lim$res_RT.tag)),c("Phylum","Class","Order","Family","Genus")])
dev.off()

ratiodf_h<-ratiodf_h[,c(3,2,5,8,4,7,10)]
colnames(ratiodf_h)<-c("OTU",paste0("RATIO.",reactors),paste0("Meanabundance.",reactors))
rownames(ratiodf_h)<-ratiodf_h$OTU
finalfram<-merge(ratiodf_h,taxinfo,by="row.names",all.x=TRUE) # Merge taxonomi and ratio dataframes
maxabundance<-apply(X = finalfram[,c("Meanabundance.Mesophilic","Meanabundance.Thermophilic","Meanabundance.THP")], MARGIN = 1,FUN = "max") # Calcuclate maximum (mean) abundance
maxratio<-apply(X = finalfram[,c("RATIO.Mesophilic","RATIO.Thermophilic","RATIO.THP")], MARGIN = 1,FUN = "max") # Calculate maximum ratio
finalframe<-cbind(finalfram,maxabundance,maxratio)            # Combine 
newdata <- finalframe[order(maxabundance,decreasing = T),]    # Order the OTUs by abundance
newdata_lim<-newdata[which(newdata$maxabundance>0),c(3:14,2)] # Remove OTUs with an abundance of 0 and select which columns to save
write.csv(newdata_lim,file = "SurveyPaper/tableS2.csv",row.names = F)
```



## Figure 5
Panelfigure with 2 heatmaps. 

a. Genus level heatmap for bacteria sorted by abundance in influent streams
b. Genus level heatmap for bacteria sorted by abudnance in digesters

```{r,fig.align="center",fig.height=13,fig.width=25}
n_taxa<-50
b_ads_sFig5_sel<- subset_samples(b_df, !(SA_ID %in% skiplist) & !(Sample_location=="Herning" & Temperature=="Mesophilic") & (Investigator!="CHJ")  & !(SEQ_ID %in% skiplist2) & order_type!="empty" & simple_var3!="empty" & Sample_type!="Effluent" & !(Dataset=="CJH" & Temperature=="THP"))



# Sorted by influent abundance
b_ads_sFig5_sel_sort<- subset_samples(b_df, !(SA_ID %in% skiplist) & !(Sample_location=="Herning" & Temperature=="Mesophilic") & (Investigator!="CHJ") &  !(SEQ_ID %in% skiplist2) & order_type %in% c("1_primary","2_surplus") & !(Dataset=="CJH" & Temperature=="THP"))

p<-amp_heatmap(b_ads_sFig5_sel_sort,tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = n_taxa,
            color.vector = c("white","red"))

gvf<-rev(levels(p$data$Display))[1:n_taxa]
sort<-rev(gvf)

p_temp<-amp_heatmap(b_ads_sFig5_sel,group = "simple_var3",tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = sort,order.y = sort,plot.na = T, color.vector = c("white","red"))

# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
p5_a<-p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank(),legend.position="none",axis.text.x = element_text(vjust=0.5,color = "black"))

# Sorted by digester abundance
b_ads_sFig5_sel_sort<- subset_samples(b_df, !(SA_ID %in% skiplist) & !(Sample_location=="Herning" & Temperature=="Mesophilic") & (Investigator!="CHJ") &  !(SEQ_ID %in% skiplist2) & simple_var3 %in% c("c) Mesophilic","d) Thermophilic") & !(Dataset=="CJH" & Temperature=="THP"))

p<-amp_heatmap(b_ads_sFig5_sel_sort,tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = n_taxa,
            color.vector = c("white","red"))

gvf<-rev(levels(p$data$Display))[1:n_taxa]
sort<-rev(gvf)

p_temp<-amp_heatmap(b_ads_sFig5_sel,group = "simple_var3",tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = sort,order.y = sort,plot.na = T, color.vector = c("white","red"))

# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
p5_b<-p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank(),legend.position="none",axis.text.x = element_text(vjust=0.5,color = "black"))


plot_grid(p5_a, p5_b, align='vh',ncol = 2, labels=c('A', 'B'))
fname<-paste0("SurveyPaper/Figure5.pdf")
hfig<-(n_taxa/50)*13
#ggsave(filename = fname,width = 25,height = hfig, useDingbats = FALSE)
```

Get stats about the number of samples pr sample type

```{r}
mytable<-sample_data(b_ads_sFig5_sel)
summary(factor(mytable$simple_var3))
```

# Supplementary figures
## Figure S1
Panelfigure with 3 panels. 

a. Coreplot for mesophilic samples
b. Coreplot for thermophilic samples
c. Rankabundance curves for the three digester types (mesophilic, thermophilic and THP)

### Figure S1a

```{r}
ps1_a<-amp_core(b_ads_sM,plot.type = "core")+theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))+theme(legend.position="none")
```

### Figure S1b

```{r}
ps1_b<-amp_core(b_ads_sT,plot.type = "core")+theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))+theme(legend.position="none")
```

### Figure S1c

```{r,fig.height=5,fig.width=5}
ps1_c<-amp_rabund(b_ads_s,group = "Temperature",plot.type = "curve")+xlim(0,1000)+theme(legend.justification=c(1,0), legend.position=c(1,0))
#ps1_c+geom_vline(xintercept =300,colour="grey", linetype = "longdash")+geom_hline(yintercept = 80,colour="grey", linetype = "longdash")
```

### Figure S1 combined

```{r,fig.align="center",fig.height=5,fig.width=15}
plot_grid(ps1_a, ps1_b, ps1_c, align='vh',ncol = 3, labels=c('A', 'B', 'C'))
#ggsave(filename = "SurveyPaper/FigureS1.pdf",width = 15,height = 5, useDingbats = FALSE)
```



## Figure S2
PCA plot of all the samples coloured by sample type

```{r ,fig.align="center",fig.height=7,fig.width=7}
b_sel<- subset_samples(b_df, simple_var3!="empty" & !(SA_ID %in% skiplist) & !(Sample_location=="Herning" & Temperature=="Mesophilic") & (Investigator!="CHJ") & !(SEQ_ID %in% skiplist2) & !(Dataset=="CJH" & Temperature=="THP") & !(Project_name %in% c("DNAextraction_optimisation_from_biogas","VTU_Case_study_-_GasMixCase_Viborg")))
# Make labels prettier
sample_data(b_sel)$simple_var3<-gsub(".) ", "", sample_data(b_sel)$simple_var3)
pS2<-amp_ordinate(b_sel,
             plot.color = "simple_var3",
             plot.group.label = "simple_var3",
             plot.group.label.size = 7)+
  theme_cowplot()+
  theme(plot.margin = unit(c(0,0,0,0), "mm"))+
  theme(legend.position="none")
pS2
fname<-paste0("SurveyPaper/FigureS2.pdf")
#ggsave(filename = fname,width = 7,height = 7, useDingbats = FALSE)
```

## Figure S3
OTUs within Gelria are different between mesophilic and thermophilic! 

```{r fig.height=7,fig.width=10}
gelria<-subset_taxa(b_ads_s, Genus=="g__Gelria")

p_temp<-amp_heatmap(gelria,tax.aggregate = "OTU",tax.show = 15,tax.add = "Genus",group = c("Temperature","Sample_location"),color.vector = c("white","red"),plot.na = T)+
  facet_grid(~ Temperature,scales="free_x",space="free")+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")
# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank())

#ggsave(filename = "SurveyPaper/FigureS3.pdf",width = 10,height = 7, useDingbats = FALSE)
```

## Figure S4
Heatmap of organisms in the surplus sludge

```{r fig.height=7,fig.width=17}
AS<-subset_samples(b_df, simple_var3=="b) Surplus sludge")

p_temp<-amp_heatmap(AS,tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = 20,group = c("Sample_location"),color.vector = c("white","red"),plot.na = T)+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")
# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank())

#ggsave(filename = "SurveyPaper/FigureS4.pdf",width = 17,height = 7, useDingbats = FALSE)
```

Get stats about the number of samples pr plant

```{r}
mytable<-sample_data(AS)
summary(factor(mytable$Sample_location))
```

## Figure S5
Heatmap of the organisms in the primary sludge

```{r, fig.height=7,fig.width=17}
PS<-subset_samples(b_df, simple_var3=="a) Primary sludge")

p_temp<-amp_heatmap(PS,tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = 20,group = c("Sample_location"),color.vector = c("white","red"),plot.na = T)+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")
# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank())

#ggsave(filename = "SurveyPaper/FigureS5.pdf",width = 17,height = 7, useDingbats = FALSE)
```

Get stats about the number of samples pr plant

```{r}
mytable<-sample_data(PS)
summary(factor(mytable$Sample_location))
```

## Figure S6
Panelfigure with 2 heatmaps. 

a. Timeseries for mesophlic digester in Randers with primary and surplus sludge samples sorted by the abundance in the influent streams
b. Timeseries for mesophlic digester in Randers with primary and surplus sludge samples sorted by the abundance in the digester

```{r,fig.align="center",fig.height=25,fig.width=20}
n_taxa<-50
b_ads_sFigs10_sel<- subset_samples(b_df, Sample_location=="Randers" & !(SEQ_ID %in% c("16SAMP-8786","16SAMP-8790","16SAMP-8794","16SAMP-8121","16SAMP-8787","16SAMP-8791","16SAMP-8795","16SAMP-8799","16SAMP-8803","16SAMP-8798","16SAMP-8802","16SAMP-8797")))



# Sorted by influent abundance
b_ads_sFigs10_sel_sort<- subset_samples(b_df, Sample_location=="Randers" & (simple_var3 %in% c("a) Primary sludge", "b) Surplus sludge")) & !(SEQ_ID %in% c("16SAMP-8786","16SAMP-8790","16SAMP-8794","16SAMP-8121","16SAMP-8787","16SAMP-8791","16SAMP-8795","16SAMP-8799","16SAMP-8803","16SAMP-8798","16SAMP-8802","16SAMP-8797")))

p<-amp_heatmap(b_ads_sFigs10_sel_sort,tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = n_taxa,
            color.vector = c("white","red"))

gvf<-rev(levels(p$data$Display))[1:n_taxa]
sort<-rev(gvf)

# make plot
p_temp<-amp_heatmap(b_ads_sFigs10_sel,group = c("Collection_date","simple_var3"),tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = sort,order.y = sort,plot.na = T, color.vector = c("white","red"))
# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub("c) ", "", original_labels))
new_labels <- toupper(sub("A) ", "", new_labels))
new_labels <- toupper(sub("B) ", "", new_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
ps10_a<-p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank())+theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")


# Sorted by digester abundance
b_ads_sFigs10_sel_sort<- subset_samples(b_df, Sample_location=="Randers" & (simple_var3 %in% c("c) Mesophilic")) & !(SEQ_ID %in% c("16SAMP-8786","16SAMP-8790","16SAMP-8794","16SAMP-8121","16SAMP-8787","16SAMP-8791","16SAMP-8795","16SAMP-8799","16SAMP-8803","16SAMP-8798","16SAMP-8802","16SAMP-8797")))

p<-amp_heatmap(b_ads_sFigs10_sel_sort,tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = n_taxa,
            color.vector = c("white","red"))

gvf<-rev(levels(p$data$Display))[1:n_taxa]
sort<-rev(gvf)

# make plot
p_temp<-amp_heatmap(b_ads_sFigs10_sel,group = c("Collection_date","simple_var3"),tax.aggregate = "Genus",tax.add = c("Phylum","Class","Order","Family"),tax.empty = "OTU",tax.show = sort,order.y = sort,plot.na = T, color.vector = c("white","red"))
# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub("c) ", "", original_labels))
new_labels <- toupper(sub("A) ", "", new_labels))
new_labels <- toupper(sub("B) ", "", new_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
ps10_b<-p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank())+theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")

plot_grid(ps10_a, ps10_b, align='vh',ncol = 1, labels=c('A', 'B'))
fname<-paste0("SurveyPaper/FigureS6.pdf")
#ggsave(filename = fname,width = 20,height = 25, useDingbats = FALSE)

```

## Figure S7
OTU level heatmap for Archaea
```{r,fig.align="center",fig.width=18,fig.height=9}
p_temp<-amp_heatmap(a_ads_s,tax.aggregate = "OTU",tax.add = c("Phylum","Class","Order","Family","Genus"),tax.empty = "OTU",tax.show = 20,group = c("Temperature","Sample_location"),color.vector = c("white","red"),plot.na = T)+
  facet_grid(~ Temperature,scales="free_x",space="free")+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")

# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank())

# Save figure
#ggsave(filename = "SurveyPaper/FigureS7.pdf",width = 18,height = 9, useDingbats = FALSE)
```

## Figure S8
OTU level heatmap for Bacteria
```{r,fig.align="center",fig.width=18,fig.height=9}
p_temp<-amp_heatmap(b_ads_s,tax.aggregate = "OTU",tax.add = c("Phylum","Class","Order","Family","Genus"),tax.empty = "OTU",tax.show = 20,group = c("Temperature","Sample_location"),color.vector = c("white","red"),plot.na = T)+
  facet_grid(~ Temperature,scales="free_x",space="free")+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")

# Change labels
original_labels<-levels(p_temp$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p_temp$data$Group.label<-mapvalues(p_temp$data$Group,original_labels,c(new_labels)) 
p_temp+aes(x=p_temp$data$Group.label)+theme(axis.title.x=element_blank())

# Save figure
#ggsave(filename = "SurveyPaper/FigureS8.pdf",width = 18,height = 9, useDingbats = FALSE)
```


## Figure S9
Heatmap with classification of OTUs based on their ratios

```{r fig.align="center",fig.width=20,fig.height=45}
newdata$RATIOGROUP<-"empty"
newdata$RATIOGROUP[which(newdata$maxratio>1)]<-"VAR" # 1
newdata$RATIOGROUP[which(newdata$maxratio>10)]<-"POS" # 10
newdata$RATIOGROUP[which(newdata$maxratio<1)]<-"NEG" # 1
ratiofuncttable<-newdata[,c("OTU","RATIOGROUP")]
row.names(ratiofuncttable) <- NULL 
names(ratiofuncttable)[1]<-"Genus"
```

```{r fig.align="center",fig.width=20,fig.height=45}
nshow<-100
p1 <- amp_heatmap(data = b_ads_s,
                  c("Temperature","Sample_location"), 
                  tax.aggregate = "OTU", 
                  tax.add=c("Phylum","Class","Order","Family","Genus"),
                  tax.show=nshow, 
                  plot.na =T,
                  plot.colorscale = "log",
                  color.vector = c("white","red")) + 
  facet_grid(~ Temperature,scales="free_x",space="free")+
  theme(plot.margin = unit(c(0,0,0,0), "mm"),
        axis.text.y = element_text(color = "black"),
        strip.text.x = element_text(color = "black"),
        axis.text.x = element_text(vjust=0.5,color = "black"),legend.position="none")
# Change labels
original_labels<-levels(p1$data$Group)
new_labels <- toupper(sub(".*? ", "", original_labels))
p1$data$Group.label<-mapvalues(p1$data$Group,original_labels,c(new_labels)) 
p1<-p1+aes(x=p1$data$Group.label)+theme(axis.title.x=element_blank())

p2<-amp_function(p1,adjust = 0,genus.pos = 6,functions = ratiofuncttable, selection = "RATIOGROUP")+theme(legend.position="none",plot.margin = unit(c(10,0,0,0), "mm"))
grid.arrange(p1, p2, nrow = 1, widths = c(0.95, 0.05))

p <- arrangeGrob(p1, p2, nrow=1,widths = c(0.95,0.05))
grid.draw(p) # interactive device works
# Save figure
#ggsave(filename = "SurveyPaper/FigureS9.pdf",plot = p,width = 20,height = 45, useDingbats = FALSE)
```

## Figure S9
Distributions of digester to influent abundance ratios for the three reactor types (Mesophilic, thermophlic and THP)
```{r}
ggplot(ratiodf)+
  geom_density(aes(x=RATIO,y=(..density..),weight=Frequency, colour = T_reactor))+
  ylab(label = "Abundance weighted density")+
  scale_x_log10(breaks = c(0.1,1,10,100,1000))+
  theme(legend.title=element_blank(),legend.position = "bottom")
# Save the figure
#ggsave(filename = "SurveyPaper/FigureS10.pdf",width = 10,height = 7, useDingbats = FALSE)
```

